pkg/scheduler/scheduler.go

```diff
@@ -72,58 +78,156 @@ func (s *Scheduler) Less(pod1, pod2 *framework.QueuedPodInfo) bool {
 	return time1.Before(time2)
 }
 
+// PreFilter is called at the beginning of the scheduling cycle. All PreFilter
+// plugins must return success or the pod will be rejected.
 func (s *Scheduler) PreFilter(ctx context.Context, state *framework.CycleState, pod *v1.Pod) *framework.Status {
 	// Do nothing
-	pg, err := s.podGroupManager.PodGroup(pod)
-	if err != nil {
-		return framework.NewStatus(framework.Error, err.Error())
-	}
-
-	if pg != nil {
-		klog.V(3).Infof("podgroup: %v", pg)
-	}
 	return framework.NewStatus(framework.Success, "")
 }
 
+// PreFilterExtensions returns a PreFilterExtensions interface if the plugin implements one,
+// or nil if it does not. A Pre-filter plugin can provide extensions to incrementally
+// modify its pre-processed info. The framework guarantees that the extensions
+// AddPod/RemovePod will only be called after PreFilter, possibly on a cloned
+// CycleState, and may call those functions more than once before calling
+// Filter again on a specific node.
 func (s *Scheduler) PreFilterExtensions() framework.PreFilterExtensions {
 	return nil
 }
 
+// Filter is called by the scheduling framework.
+// All FilterPlugins should return "Success" to declare that
+// the given node fits the pod. If Filter doesn't return "Success",
+// it will return "Unschedulable", "UnschedulableAndUnresolvable" or "Error".
+// For the node being evaluated, Filter plugins should look at the passed
+// nodeInfo reference for this particular node's information (e.g., pods
+// considered to be running on the node) instead of looking it up in the
+// NodeInfoSnapshot because we don't guarantee that they will be the same.
+// For example, during preemption, we may pass a copy of the original
+// nodeInfo object that has some pods removed from it to evaluate the
+// possibility of preempting them to schedule the target pod.
+func (s *Scheduler) Filter(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
```

Comment:
When I check the feature, I find that the current implementation is smart 👍.

It avoids us from using taint and tolerations to isolate pods of different prod groups otherwise a Controller is needed to remove the taint after the pod group is deleted.

Reply:
😃